packages/rancher-sriov/generated-changes/patch/templates/clusterrole.yaml.patch

```diff
@@ -0,0 +1,61 @@
+--- charts-original/templates/clusterrole.yaml
++++ charts/templates/clusterrole.yaml
+@@ -15,13 +15,13 @@
+     resources: ["daemonsets"]
+     verbs: ["get"]
+   - apiGroups: [""]
+-    resources: [namespaces, serviceaccounts]
++    resources: ["namespaces", "serviceaccounts"]
+     verbs: ["*"]
+   - apiGroups: ["k8s.cni.cncf.io"]
+     resources: ["network-attachment-definitions"]
+     verbs: ["*"]
+   - apiGroups: ["rbac.authorization.k8s.io"]
+-    resources: [clusterroles, clusterrolebindings]
++    resources: ["clusterroles", "clusterrolebindings"]
+     verbs: ["*"]
+   - apiGroups: ["admissionregistration.k8s.io"]
+     resources: ["mutatingwebhookconfigurations", "validatingwebhookconfigurations"]
+@@ -52,3 +52,42 @@
+   - apiGroups: [""]
+     resources: ["pods/eviction"]
+     verbs: ["create"]
++---
++apiVersion: rbac.authorization.k8s.io/v1
++kind: ClusterRole
++metadata:
++  name: sriov-admin
```

Comment:
Do we expect these clusterroles to automatically aggregate into the default k8s admin, edit, and view roles? If so, we need annotations here, e.g.

https://github.com/rancher/charts/blob/7836aff0c1b2846c3fbcf8ed2adc6963d21c20b1/charts/rancher-monitoring/rancher-monitoring/14.5.101%2Bup14.5.0/templates/rancher-monitoring/clusterrole.yaml#L7-L9

Reply:
Not sure about this. I added this roles because it was required based on https://confluence.suse.com/pages/viewpage.action?spaceKey=EN&title=Feature+Charts+Helm+Requirements, i.e. we don't need them to run this chart correctly. Do you normally recommend to add that to charts?

Reply:
So the reason why these annotations are necessary is because they [aggregate these ClusterRoles](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#aggregated-clusterroles) to the [default k8s roles](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#user-facing-roles), which are bound to corresponding Rancher role ([Project Owner -> admin](https://github.com/rancher/rancher/blob/9c9491b2879de82454ab53d083492e530e519693/pkg/data/management/role_data.go#L229), [Project Member -> edit](https://github.com/rancher/rancher/blob/9c9491b2879de82454ab53d083492e530e519693/pkg/data/management/role_data.go#L265), [Read-only -> view](https://github.com/rancher/rancher/blob/9c9491b2879de82454ab53d083492e530e519693/pkg/data/management/role_data.go#L297)).

This is a choice made for every chart, so I think it depends on your RBAC expectations for this chart. Do you expect the person who owns or is part of the project that contains `cattle-sriov-system` to have these permissions by default, or should they be manually bound as and when necessary?

cc: @cbron if you have any thoughts.

Reply:
Thanks for the explanation. I can see users having only "sriov-view" permissions, because this chart modifies the network configuration of the host and that can create connectivity issues across nodes. Therefore, only the "admin" of the cluster should be able to have full control of this. I guess these should be made available by default. Let me add those